src/julia_gc.c

```diff
@@ -491,7 +522,7 @@ static void SafeScanTaskStack(PtrArray * stack, void * start, void * end)
         // longjmps to JuliaTLS->safe_restore; we use this
         // mechamism to abort stack scanning when a protected
         // page is hit. For this to work, we must scan the stack
-        // from top to bottom, so we see any guard pages last.
+        // from bottom to top, so we see any guard pages last.
```

Comment:
this doesn't fit the terminology above anymore
```suggestion
        // from start to end (as defined above), so we see any guard pages last.
```

Reply:
Umm, I don't see it or maybe I'm misunderstanding something? The guard pages are at the start of the buffer. I agree that "to top" is misleading and should be "towards the start of the stack buffer" or something like that. But the direction is high -> low address?

Reply:
The thing is that in here, we don't necessary scan till the top (as we don't know where it is). As to whether we scan from "end to start" or "start to end" , I forgot that we scan "downwards"; so we could say "from end downwards to the start". But it would indeed nice to say "from bottom to top"; alas, again, I am concerned that "top" is not accurate here in general?

We could also think about calling `start` something else, like `max_top` or `absolute_top` or so?

Reply:
So, I've rephrased comments to refer to the start/end of the *stack buffer* that contains the stack, and the top/bottom of the *stack* and also did some rewording to describe the actual process.